card-tricks/ena/card_tricks.go

```diff
@@ -0,0 +1,43 @@
+package cards
+
+func indexInBound(slice []int, index int) bool {
+	return index >= 0 && index < len(slice)
+}
+
+func GetItem(slice []int, index int) (int, bool) {
+	if indexInBound(slice, index) {
+		return slice[index], true
+	}
+	return 0, false
+}
+
+func SetItem(slice []int, index, value int) []int {
+	if indexInBound(slice, index) {
+		slice[index] = value
+	} else {
+		slice = append(slice, value)
+	}
+	return slice
+}
+
+func PrefilledSlice(value, length int) []int {
+	if length < 1 {
+		return []int{}
+	}
+	ret := make([]int, length)
+	for i, _ := range ret {
+		ret[i] = value
+	}
+	return ret
+}
+
+func RemoveItem(slice []int, index int) []int {
+	if indexInBound(slice, index) {
+		if len(slice) == index-1 {
+			slice = slice[:index]
```

Comment:
You don't need this part. You only have to check that the index is in bounds and then do what you are doing in line 39.

Reply:
@David: Wouldn't slice[index+1:] result in an array out of bounds if the index is the last one in the slice?

Reply:
Nope https://go.dev/play/p/0pR_ouPcGsc

Reply:
Yes, it would https://go.dev/play/p/JBWIha8TZZJ

Reply:
Weird, slice[index+1] gives out of bound but slice[index+1:] does not because “slicingl can start from len() and end in len()

Reply:
@enarha that index is already out of bounds (the slice there has to elements, so the maximum value for index is 1). You have to validate the index obviously, but `[:index+1]` will not fail provided that.

Reply:
ah, sorry, it does not. I actually checked that for strings, but my test with []int returned out of band. Probably I did the same test with specific index and not range.

Reply:
Right, if the index in the call is really out of bounds then you get the error.  It could make sense making removing something out of bounds be a no-op like you wrote, but in the real world there should be also and err returned for those cases.

I still prefer the original code, just my opinion though.